contracts/Swap/SwapRouter.sol

```diff
@@ -14,40 +17,54 @@ import "./interfaces/IVBNB.sol";
  * @author 0xlucian
  */
 
-contract SwapRouter is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, RouterHelper, IPancakeSwapV2Router {
+contract SwapRouter is Ownable2StepUpgradeable, RouterHelper, IPancakeSwapV2Router {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    address public comptrollerAddress;
+
     // ***************
     // ** MODIFIERS **
     // ***************
-
     modifier ensure(uint256 deadline) {
         if (deadline < block.timestamp) {
             revert SwapDeadlineExpire(deadline, block.timestamp);
         }
         _;
     }
 
-    receive() external payable {
-        assert(msg.sender == WBNB); // only accept BNB via fallback from the WBNB contract
+    modifier ensureVTokenListed(address vTokenAddress) {
+        bool isListed = InterfaceComptroller(comptrollerAddress).markets(vTokenAddress);
+        if (isListed != true) {
+            revert VTokenNotListed(vTokenAddress);
+        }
+        _;
     }
 
+    /// @notice event emitted on sweep token success
+    event SweepToken(address token, address to, uint256 sweepAmount);
+
     // *********************
     // **** CONSTRUCTOR ****
     // *********************
 
     /// @notice Constructor for the implementation contract. Sets immutable variables.
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(address WBNB_, address factory_) RouterHelper(WBNB_, factory_) {
+    constructor(address WBNB_, address factory_, address _comptrollerAddress) RouterHelper(WBNB_, factory_) {
```

Comment:
We have to update the deployment script to provide this address: `deploy/005-deploy-swaprouter.ts`